src/main/java/ca/teamdman/sfm/common/program/SimulateExploreAllPathsProgramBehaviour.java

```diff
@@ -184,7 +181,7 @@ public record Branch(
     public record IO(
             IOStatement statement,
             IOKind kind,
-            Set<ResourceType<?,?,?>> usedResourceTypes,
+            Collection<ResourceType<?,?,?>> usedResourceTypes,
```

Comment:
```suggestion
            Set<ResourceType<?,?,?>> usedResourceTypes,
```
Set is used to enforce uniqueness